### ISSUE TITLE
(python) cv2.resize() return error

### ISSUE THREAD

**Issue by JiaPai12138** [opencv/opencv#21102]
##### System information (version)

- OpenCV => 4.5.4
- Operating System / Platform => win10 LTSC 2021
- Compiler => Visual Studio 2019
- Python Version => 3.9.9

##### Detailed description

- cv2.resize() function return cv2.error (-215:Assertion failed) _step >= minstep in function 'cv::Mat::Mat'
- opencv was compiled with cuda version 11.4.3, cudnn version 8.2.4 following tutorial from [YouTube](https://www.youtube.com/watch?v=HsuKxjQhFU0)

##### Steps to reproduce

```
import cv2
import numpy as np
origin_black = np.zeros((1, 1, 3), np.uint8)
cv2.resize(origin_black, (100, 100))
```
    (then it returns) cv2.error: OpenCV(4.5.4) D:\a\opencv-python\opencv-python\opencv\modules\core\src\matrix.cpp:466: error: (-215:Assertion failed) _step >= minstep in function 'cv::Mat::Mat'

##### Issue submission checklist

 - [ ] I report the issue, it's not a question
   
 - [ ] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [ ] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by JiaPai12138** [comment]
My bad, np.zeros([1,1,3],dtype=np.uint8) would work

**Comment by DumDereDum** [comment]
In `resize` function step doesn't change in the case when cv::Mat has only one element ( like [[[0]]] ).
Step equal 1 all the time despite the fact that size of the only one element mote than 1 ( like [[[0 0]]] ).

In the table the results of my tests
Size(x, y, z) ~ np.zeros([x, y, z], np.unit8)

| Size(x, y, z) | [1, 1, 1] | [1, 1, 2] | [1, 1, 3] | [1, 2, 1] | [1, 2, 2] | [1, 2, 3] |
| ------------- | --------- | --------- | --------- | --------- | --------- | --------- |
| esz           | 1         | 2         | 3         | 1         | 2         | 3         |
| type          | 0         | 8         | 16        | 0         | 8         | 16        |
| step          | 1         | 1         | 1         | 2         | 4         | 6         |
| minstep       | 1         | 2         | 3         | 2         | 4         | 6         |

I guess that `step == minstep` in the case of one element could be a possible solution for this issue
Following [code](https://github.com/opencv/opencv/blob/4.x/modules/core/src/matrix.cpp#L451-L478)

**Comment by nimishbongale** [comment]
Can I take this up @DumDereDum?

**Comment by DumDereDum** [comment]
fixed in OpenCV 4.5.5

**Comment by alalek** [comment]
duplicate of #20968
